franklin/builder/models.py

```diff
@@ -34,14 +34,28 @@ def build(self):
                     "path": self.path,
                     "repo_name": repo_name
                 }
-        r = requests.post(url, data=json.dumps(body), headers=headers)
-        response = r.json()['building']
-        if not response:
-            logger.error("Builder says its not building")
-            print('Response HTTP Status Code   : {status_code}'.format(
-                status_code=r.status_code))
-            print('Response HTTP Response Body : {content}'.format(
-                content=r.content))
+        r = None
+        try:
+            r = requests.post(url, data=json.dumps(body), headers=headers)
+        except (ConnectionError, HTTPError, Timeout) as e:
+            logger.error('Response HTTP Status Code   : %s', e.status_code)
+            logger.error('Response HTTP Response Body : %s', e.content)
+        except:
+            logger.error('Unexpected Builder error: %s', sys.exc_info()[0])
```

Comment:
Apparently %s is the most `pythonic` way of constructing these types of logs. http://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html